HW2/P3/P3.txt

```diff
@@ -0,0 +1,153 @@
+
+
+Multithreading
+
+num threads:  1
+1074.652832 Million Complex FMAs in 5.04761886597 seconds, 212.902927209 million Complex FMAs / second
+
+num threads:  2
+1074.652832 Million Complex FMAs in 2.54001998901 seconds, 423.088336568 million Complex FMAs / second
+
+num threads:  4
+1074.652832 Million Complex FMAs in 1.29837203026 seconds, 827.692531074 million Complex FMAs / second
+
+Discussion: As expected, the runtime of the mandelbrot computations gets reduced by about a factor of 2
+as the number of threads doubles. Since the threads run independently in parallel, the runtime is 
+inversely proportional to the number of threads and the speed up is not exactly 2 due to message 
+passing and other overhead. 
+
+Instruction-level Parallelism
+
+num threads:  1
+1074.94829 Million Complex FMAs in 1.15263199806 seconds, 932.60320016 million Complex FMAs / second
+
+num threads:  2
+1074.94829 Million Complex FMAs in 0.648010969162 seconds, 1658.84273748 million Complex FMAs / second
+
+num threads:  4
+
+1074.94829 Million Complex FMAs in 0.346281051636 seconds, 3104.26540789 million Complex FMAs / second
+
```

Comment:
I would expect a greater speedup - closer to 20x with 4 threads and AVX as opposed to the 15x here